### ISSUE TITLE
Unable to verify tokens when using bas64 encoded secrets

### ISSUE THREAD

**Issue by EJMason** [nestjs/jwt#94]
## Current behavior

 It looks like tokens cannot be verified if the secret is Base64 encoded. The issue is caused by the implementation of the verify method. [Link to Method](https://github.com/nestjs/jwt/blob/715f2aaa85073d42e96c5631a865904e901b8bcb/lib/jwt.service.ts#L54-L64)

```javascript
  verify<T extends object = any>(
    token: string,
    options?: jwt.VerifyOptions
  ): T {
    const verifyOptions = this.mergeJwtOptions(options, 'verifyOptions');
    const secret = this.getSecretKey(
      token,
      options,
      'publicKey',
      JwtSecretRequestType.VERIFY
    );

    return jwt.verify(token, secret.toString(), verifyOptions) as T;
  }
```
The issue is being caused by the stringification of the secret in this line: 

```javascript 
return jwt.verify(token, secret.toString(), verifyOptions) as T;
```

Check out the [typescript definitions](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/4f60df3ab37a512ff52377403a7ac75c14b466f5/types/jsonwebtoken/index.d.ts#L118-L121) for the jsonwebtoken library, the underlying library for nestjs/jwt:

```javascript
export type Secret =
    | string
    | Buffer
    | { key: string | Buffer; passphrase: string };
```
And here is the Typescript definition for the verify implementation:

```javascript
export function verify(
    token: string,
    secretOrPublicKey: string | Buffer,
    options?: VerifyOptions,
): object | string;
```

And here are the type definitions for the nestjs/jwt:

```javascript
export interface JwtModuleOptions {
  signOptions?: jwt.SignOptions;
  secret?: string | Buffer;   // <-------  THIS ONE HERE
  publicKey?: string | Buffer;
  privateKey?: jwt.Secret;  // <----- ALSO, THIS ONE HERE
  /**
   * @deprecated
   */
  secretOrPrivateKey?: jwt.Secret;
  secretOrKeyProvider?: (
    requestType: JwtSecretRequestType,
    tokenOrPayload: string | object | Buffer,
    options?: jwt.VerifyOptions | jwt.SignOptions
  ) => jwt.Secret;
  verifyOptions?: jwt.VerifyOptions;
}
```

The verify method CAN take a buffer but it is being stringified here. This causes keys that are Base64 encoded to fail always when used with the nestjs/jwt package.

## Expected behavior

JWT should be verified correctly if the the secret is of Buffer type. The buffer should not be using toString() in the verify method.

## Minimal reproduction of the problem with instructions

[Gist of code to verify problematic code](https://gist.github.com/EJMason/1cb622cbaf1e9978caa287ed3c8d0e38)

## What is the motivation / use case for changing the behavior?

We would love to stick to the nest ecosystem, but our company uses base64 buffers for creating JWT's. 

## Environment

<pre><code>
Nest version: 6.7.2

 
For Tooling issues:
- Node version: 10.15.1  
- Platform:  Mac 

Others:

</code></pre>

**Comment by EJMason** [comment]
Sure thing! Did you envision anything other than removing the toString call? My plan was to remove it and add some unit tests. Was the toString motivated by any edge case that I could also address in the PR?

**Comment by EJMason** [comment]
@kamilmysliwiec Also, one more thing. Looks like there is a method overload in the jsonwebtoken library that allows users to pass a callback instead of a private key as a second parameter for the verify method. This enables the ability to fetch keys from an external source instead of keeping them in state. 

```typescript
// Verify using getKey callback
// Example uses https://github.com/auth0/node-jwks-rsa as a way to fetch the keys.
var jwksClient = require('jwks-rsa');
var client = jwksClient({
  jwksUri: 'https://sandrino.auth0.com/.well-known/jwks.json'
});
function getKey(header, callback){
  client.getSigningKey(header.kid, function(err, key) {
    var signingKey = key.publicKey || key.rsaPublicKey;
    callback(null, signingKey);
  });
}
```

Check out the typescript definitions for the 2 overloads:

```typescript
/**
 * Asynchronously verify given token using a secret or a public key to get a decoded token
 * token - JWT string to verify
 * secretOrPublicKey - A string or buffer containing either the secret for HMAC algorithms,
 * or the PEM encoded public key for RSA and ECDSA. If jwt.verify is called asynchronous,
 * secretOrPublicKey can be a function that should fetch the secret or public key
 * [options] - Options for the verification
 * callback - Callback to get the decoded token on
 */
export function verify(
    token: string,
    secretOrPublicKey: string | Buffer | GetPublicKeyOrSecret,
    callback?: VerifyCallback,
): void;
export function verify(
    token: string,
    secretOrPublicKey: string | Buffer | GetPublicKeyOrSecret,
    options?: VerifyOptions,
    callback?: VerifyCallback,
): void;
```
And here is the type for the GetPublicKeyOrSecret

```typescript
export type GetPublicKeyOrSecret = (
    header: JwtHeader,
    callback: SigningKeyCallback
) => void;
```

This functionality didn't exist in the Nestjs/jwt package, but might be worth calling out in the docs that you currently can't do this. In the future this could be an added feature that can simplify this behavior using the nest api. Food for thought.... 🤔

**Comment by kamilmysliwiec** [comment]
Fixed in 6.1.2
